flash/data/data_module.py

```diff
@@ -0,0 +1,245 @@
+        return DataLoader(
+            self._train_ds if isinstance(self._train_ds, Dataset) else self._train_ds(),
+            batch_size=self.batch_size,
+            shuffle=True,
+            num_workers=self.num_workers,
+            pin_memory=True,
+            drop_last=True,
+        )
+
+    def _val_dataloader(self) -> DataLoader:
+        return DataLoader(
+            self._valid_ds if isinstance(self._valid_ds, Dataset) else self._valid_ds(),
+            batch_size=self.batch_size,
+            num_workers=self.num_workers,
+            pin_memory=True,
+        )
+
+    def _test_dataloader(self) -> DataLoader:
+        return DataLoader(
+            self._test_ds if isinstance(self._test_ds, Dataset) else self._test_ds(),
+            batch_size=self.batch_size,
+            num_workers=self.num_workers,
+            pin_memory=True,
+        )
+
+    def _predict_dataloader(self) -> DataLoader:
+        predict_ds = self._predict_ds if isinstance(self._predict_ds, Dataset) else self._predict_ds()
+        return DataLoader(
+            predict_ds,
+            batch_size=min(self.batch_size,
+                           len(predict_ds) if len(predict_ds) > 0 else 1),
+            num_workers=self.num_workers,
+            pin_memory=True,
+        )
+
+    def generate_auto_dataset(self, *args, **kwargs):
+        if all(a is None for a in args) and len(kwargs) == 0:
+            return None
+        return self.data_pipeline._generate_auto_dataset(*args, **kwargs)
+
+    @property
+    def preprocess(self) -> Preprocess:
+        return self.preprocess_cls()
+
+    @property
+    def postprocess(self) -> Postprocess:
+        return self.postprocess_cls()
+
+    @property
+    def data_pipeline(self) -> DataPipeline:
+        return DataPipeline(self.preprocess, self.postprocess)
+
+    @classmethod
+    def autogenerate_dataset(
+        cls,
+        data: Any,
+        running_stage: RunningStage,
+        whole_data_load_fn: Optional[Callable] = None,
+        per_sample_load_fn: Optional[Callable] = None,
+        data_pipeline: Optional[DataPipeline] = None,
+    ) -> AutoDataset:
+
+        if whole_data_load_fn is None:
+            whole_data_load_fn = getattr(
+                cls.preprocess_cls,
+                DataPipeline._resolve_function_hierarchy('load_data', cls.preprocess_cls, running_stage, Preprocess)
+            )
+
+        if per_sample_load_fn is None:
+            per_sample_load_fn = getattr(
+                cls.preprocess_cls,
+                DataPipeline._resolve_function_hierarchy('load_sample', cls.preprocess_cls, running_stage, Preprocess)
+            )
+        return AutoDataset(data, whole_data_load_fn, per_sample_load_fn, data_pipeline, running_stage=running_stage)
+
+    @staticmethod
+    def train_valid_test_split(
+        dataset: torch.utils.data.Dataset,
+        train_split: Optional[Union[float, int]] = None,
+        valid_split: Optional[Union[float, int]] = None,
```

Comment:
Is it a bad practice to have 0 as default?

Reply:
Why? if you don't provide any split values, the whole data will be used for training. Makes sense to me.